### PR TITLE
Bump to pytorch 25.11

### PR DESCRIPTION
**Pull request by chtruong814**
# What does this PR do ?

Bump to pytorch 25.11

**Collection**: [Note which collection this PR will affect]

# Changelog

- Add specific line by line info of high level changes in this PR.

# Usage

- You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"

**Pre checks**:

- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:

- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed.
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information

- Related to # (issue)

### CODE REVIEW

File: nemo/collections/common/data/lhotse/__init__.py [hunk 1]
```diff
@@ -25,3 +29,14 @@
...
+
+# Newer versions of torch.utils.data.Sampler do not have the data_source parameter.
+# lhotse's CutSampler expects to pass data_source to Sampler, so we need to patch
+# Sampler.__init__ to accept it.
+if "data_source" not in inspect.signature(Sampler.__init__).parameters:
+
+    def patched_sampler_init(self, data_source=None):
+        pass
+
+    Sampler.__init__ = patched_sampler_init
```
Comment (nithinraok) [inline nemo/collections/common/data/lhotse/__init__.py:42]: @pzelasko

Comment (pzelasko) [reply nemo/collections/common/data/lhotse/__init__.py:42]: @chtruong814 How urgent is it to merge this PR? I can release a new lhotse version tested against latest pytorch so that this patch is not needed.

Comment (chtruong814) [reply nemo/collections/common/data/lhotse/__init__.py:42]: We're going to build a container release candidate at end of Monday.  Do you think we can do that before then?  If not, we can merge this and then remove when the updated lhotse is ready?

Comment (pzelasko) [reply nemo/collections/common/data/lhotse/__init__.py:42]: OK if it passes all tests, let's do it and remove later.
